### PR TITLE
Project Hierarchy module

### PR DESCRIPTION
**Pull request by ludwiktrammer**

### CODE REVIEW

File: addons/protected_fields/models.py [hunk 1]
```diff
@@ -52,12 +52,13 @@ def _check_field_permissions(self, vals, permitted_vals=None):
...
+        record._check_field_permissions(vals, permitted_vals=defaults)
         return record
 
     @api.multi
     def write(self, vals):
         self._check_field_permissions(vals)
-        record = super(ProtectedFieldsMixin, self).write(vals)
-        return record
+        success = super(ProtectedFieldsMixin, self).write(vals)
+        self._check_field_permissions(vals)
```
Comment (ludwiktrammer) [inline addons/protected_fields/models.py]: Is it really scary? It's all performed in the same Postgresql transaction, which is automatically rolled back if the permission test fails. Also note that the test is performed twice - both before and after the write (the first one checks if the user has permissions to change the fields at all, the second one ensures the object wasn't changed beyond those permissions).

Comment (ludwiktrammer) [inline addons/protected_fields/models.py]: But if you feel uncomfortable with this solution I can do all this ("save" and check) in draft mode (changes made in draft mode are never send to the db) and only then performed another save - this time outside of the draft mode. Would it be better?

Comment (KrzysiekJ) [inline addons/protected_fields/models.py]: No, it was my bad, your first solution is 100% fine. :)

**Comment by KrzysiekJ** [comment]
Merged in 4ec81cd.

**Pull request closed without merging**